clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1179,8 +1179,14 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     Builder.defineMacro("__SYCL_DEVICE_ONLY__", "1");
     Builder.defineMacro("SYCL_EXTERNAL", "__attribute__((sycl_device))");
 
-    if (TI.getTriple().isNVPTX()) {
-        Builder.defineMacro("__SYCL_NVPTX__", "1");
+    const llvm::Triple &DeviceTriple = TI.getTriple();
+    if (DeviceTriple.isNVPTX()) {
+      Builder.defineMacro("__SYCL_NVPTX__", "1");
+    }
+    const llvm::Triple::SubArchType DeviceSubArch = DeviceTriple.getSubArch();
+    if (DeviceTriple.isSPIR() &&
+        DeviceSubArch != llvm::Triple::SPIRSubArch_fpga) {
+      Builder.defineMacro("SYCL_USE_NATIVE_FP_ATOMICS");
     }
```

Comment:
```suggestion
    if (DeviceTriple.isSPIR() &&
        DeviceTriple.getSubArch() != llvm::Triple::SPIRSubArch_fpga)
      Builder.defineMacro("SYCL_USE_NATIVE_FP_ATOMICS");
```
This is defining a macro that's not a reserved identifier -- do we want to use `__SYCL_USE_NATIVE_FP_ATOMICS__` instead?

Reply:
Until now, the feature users were actually defining the macro themselves - and may want to undefine it in some cases via the `-U` command line option. Since it's also a name that is currently used within DPC++ headers, it could be best to retain the naming within this PR, possibly changing the name after all device targets support the native atomics and explicit macro usage becomes impractical.

Reply:
Ah, good to know, thanks! FWIW, I've been beating on the "we need to use reserved identifiers in the SYCL headers" drum a fair amount lately, so I bring up the question almost reflexively at this point. :-D